enterprise/internal/batches/types/changeset_spec.go

```diff
@@ -157,11 +157,17 @@ func (cs *ChangesetSpec) computeDiffStat() error {
 
 // computeForkNamespace calculates the namespace that the changeset spec will be
 // forked into, if any.
-func (cs *ChangesetSpec) computeForkNamespace() {
+func (cs *ChangesetSpec) computeForkNamespace(forkFromSpec *bool) {
 	// Right now, we only look at the global enforceForks setting, but we will
 	// likely base this off the description eventually as well.
-	if conf.Get().BatchChangesEnforceForks {
-		cs.setForkToUser()
+	if forkFromSpec == nil {
+		if conf.Get().BatchChangesEnforceForks {
+			cs.setForkToUser()
+		}
+	} else {
+		if *forkFromSpec {
+			cs.setForkToUser()
+		}
 	}
```

Comment:
Do you think we can simplify this further thus?

```suggestion
    isBatchChangesForksEnforced := forkFromSpec == nill && conf.Get().BatchChangesEnforceForks
    isForkEnabledInSpec := forkFromSpec != nill && *forkFromSpec
   
	if isBatchChangesForksEnforced || isForkEnabledInSpec {
		cs.setForkToUser()
	}
```

That way, it's easier to read and understand. What do you think?

Reply:
In my opinion this was a little harder to read and I also believe having the nested cases leaves this open to be refactored for when we decide to support other values beside true and false. What do you think?

Reply:
No strong feelings here but I'd lean towards Adeola's original structure, too, personally.

Reply:
Yeah, it's only a matter of preference since both branches call the same function.
You can ignore this @adeola-ak , let's get this merged 🎉